snuba/datasets/discover.py

```diff
@@ -263,47 +192,41 @@ def __init__(self) -> None:
             ]
         )
 
+        # TODO: Move the storage definition out of events and transactions dataset so
+        # we can reuse the class itself and avoid the isinstance call.
+        events_dataset = get_dataset(EVENTS)
+        assert isinstance(events_dataset, EventsDataset)
+        transactions_dataset = get_dataset(TRANSACTIONS)
+        assert isinstance(transactions_dataset, TransactionsDataset)
+        storage_selector = DiscoverQueryStorageSelector(
```

Comment:
This is horrible but it is removed in PR #829 otherwise this PR would have been twice as long just by moving code around.